python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -102,6 +102,23 @@ def dist_inference(self, g, get_input_embeds, batch_size, fanout,
         return dist_inference(g, self, get_input_embeds, batch_size, fanout,
                             edge_mask=edge_mask, task_tracker=task_tracker)
 
+def prepare_for_wholegraph(g, input_nodes):
```

Comment:
Can you write a unit test for this function?

Reply:
Added.